embulk-input-s3/src/main/java/org/embulk/input/s3/ClientConfigurationConfigurable.java

```diff
@@ -0,0 +1,421 @@
+import org.embulk.config.Task;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.SecureRandom;
+
+/**
+ * Created by takahiro.nakayama on 3/25/16.
+ * This class is utility class for ClientConfiguration
+ */
+public abstract class ClientConfigurationConfigurable
+{
+    public interface Task
+        extends org.embulk.config.Task
+    {
+        @Config("protocol")
+        @ConfigDefault("null")
+        Optional<Protocol> getProtocol();
+
+        @Config("max_connections")
+        @ConfigDefault("null") // default: 50
+        Optional<Integer> getMaxConnections();
+
+        @Config("user_agent")
+        @ConfigDefault("null")
+        Optional<String> getUserAgent();
+
+        @Config("local_address")
+        @ConfigDefault("null")
+        Optional<String> getLocalAddress();
+
+        @Config("proxy_host")
+        @ConfigDefault("null")
+        Optional<String> getProxyHost();
+
+        @Config("proxy_port")
+        @ConfigDefault("null")
+        Optional<Integer> getProxyPort();
+
+        @Config("proxy_username")
+        @ConfigDefault("null")
+        Optional<String> getProxyUsername();
+
+        @Config("proxy_password")
+        @ConfigDefault("null")
+        Optional<String> getProxyPassword();
+
+        @Config("proxy_domain")
+        @ConfigDefault("null")
+        Optional<String> getProxyDomain();
+
+        @Config("proxy_workstation")
+        @ConfigDefault("null")
+        Optional<String> getProxyWorkstation();
+
+        // NOTE: RetryPolicy is a interface
+        // @Config("retry_policy")
+        // @ConfigDefault("null")
+        // Optional<RetryPolicy> getRetryPolicy();
+
+        @Config("max_error_retry")
+        @ConfigDefault("null") // default: 3
+        Optional<Integer> getMaxErrorRetry();
+
+        @Config("socket_timeout")
+        @ConfigDefault("null") // default: 8*60*1000
+        Optional<Integer> getSocketTimeout();
+
+        @Config("connection_timeout")
+        @ConfigDefault("null")
+        Optional<Integer> getConnectionTimeout();
+
+        @Config("request_timeout")
+        @ConfigDefault("null")
+        Optional<Integer> getRequestTimeout();
+
+        // NOTE: Can use `client_execution_timeout` from v1.10.65
+        // @Config("client_execution_timeout")
+        // @ConfigDefault("null")
+        // Optional<Integer> getClientExecutionTimeout();
+
+        @Config("use_reaper")
+        @ConfigDefault("null")
+        Optional<Boolean> getUseReaper();
+
+        // NOTE: Can use `use_throttle_retries` from v1.10.65
+        // @Config("use_throttle_retries")
+        // @ConfigDefault("null")
+        // Optional<Boolean> getUseThrottleRetries();
+
+        @Config("use_gzip")
+        @ConfigDefault("null")
+        Optional<Boolean> getUseGzip();
+
+        @Config("socket_send_buffer_size_hints") // used by SocketBufferSizeHints
```

Comment:
isn't this "hint" rather than "hints"? setting send buffer size and recv buffer size is plural but setting either of them is singular.